src/main/java/com/yahoo/memory/AllocateDirectMap.java

```diff
@@ -60,6 +61,9 @@
    * @throws IOException file not found or RuntimeException, etc.
    */
   static AllocateDirectMap map(final ResourceState state) throws IOException {
+    if (isFileReadOnly(state.getFile())) {
```

Comment:
If this is `AllocateDirectMap` (as opposed to `AllocateDirectWritableMap`, shouldn't it unconditionally do something like
```java
if (!state.isReadOnly()) {
  throw new IllegalStateException();
}
```
instead here?

Reply:
Or unconditionally set `state.setResourceReadOnly();`

Reply:
My goal is to make the resourceIsReadOnly_ flag a simple final (eliminating the StepBoolean), and I'm not there yet.  

What you see is correct in the current state of changes.  But in order to make this resource flag final this test and set needs to be moved earlier to when the ResourceState is first created.  This flag is strictly associated with the resource.  Each BaseWritable*Impl has its own RO flag set essentially by user choice of method calls.  The two flags are combined [here](https://github.com/DataSketches/memory/blob/NewReadOnlyScheme/src/main/java/com/yahoo/memory/WritableBuffer.java#L43) and [here](https://github.com/DataSketches/memory/blob/NewReadOnlyScheme/src/main/java/com/yahoo/memory/WritableMemory.java#L46).